include/myst/thread.h

```diff
@@ -370,8 +374,25 @@ MYST_INLINE void myst_thread_queue_push_back(
     queue->back = thread;
 }
 
-MYST_INLINE myst_thread_t* myst_thread_queue_pop_front(
+MYST_INLINE void myst_thread_queue_push_back(
+    myst_thread_queue_t* queue,
+    myst_thread_t* thread)
+{
+    myst_thread_queue_push_back_bitset(queue, thread, FUTEX_BITSET_MATCH_ANY);
+}
+
+MYST_INLINE myst_thread_t* myst_thread_queue_get_front(
```

Comment:
I don't see the need for this function. Also it introduces a branch that was not in the original code. It is used in contexts where the queue is known to be non-null.